pkg/polylog/polyzero/logger_test.go

```diff
@@ -0,0 +1,369 @@
+			Value:                  []any{"key1", "value1", "key2", 42},
+			ExpectedOutputContains: `"key1":"value1","key2":42`,
+		},
+	}
+
+	for _, level := range polyzero.Levels() {
+		testpolylog.RunEventMethodTests(
+			t,
+			level,
+			tests,
+			newTestLogger,
+			newTestEventWithLevel,
+			"*polyzero.zerologEvent",
+			getExpectedLevelOutputContains,
+		)
+	}
+}
+
+func TestZerologLogger_Levels_Discard(t *testing.T) {
+	// Construct a logger with each level. With each logger, log an event at each
+	// level and assert that the event is logged if and only if the event level
+	// is GTE the logger level.
+	for _, loggerLevel := range polyzero.Levels() {
+		testDesc := fmt.Sprintf("%s level logger", loggerLevel.String())
+		t.Run(testDesc, func(t *testing.T) {
+			logger, logOutput := newTestLogger(t, loggerLevel)
+
+			// Log an event for each level.
+			for _, eventLevel := range polyzero.Levels() {
+				event := newTestEventWithLevel(t, logger, eventLevel)
+				// Log the event level string.
+				event.Msg(eventLevel.String())
+
+				// If the event level is GTE the logger level, then the event should
+				// be logged.
+				if eventLevel.Int() >= loggerLevel.Int() {
+					require.Truef(t, event.Enabled(), "expected event to be enabled")
+					require.Contains(t, logOutput.String(), eventLevel.String())
+				} else {
+					require.Falsef(t, event.Enabled(), "expected event to be discarded")
+					require.NotContains(t, logOutput.String(), eventLevel.String())
+				}
+			}
+
+			// Print log output for manual inspection.
+			t.Log(logOutput.String())
+		})
+	}
+}
+
+func TestZerologLogger_Func_Discard_Enabled(t *testing.T) {
+	for _, loggerLevel := range polyzero.Levels() {
+		testDesc := fmt.Sprintf("%s loggerLevel logger", loggerLevel.String())
+		t.Run(testDesc, func(t *testing.T) {
+			var (
+				notExpectedOutput = "if you're reading this, the test failed"
+				// Construct a spy which implements a #Fn() method which we can use to
+				// assert that the function passed to polylog.Event#Func() is called with
+				// the expected arg(s).
+				logger, logOutput = newTestLogger(t, loggerLevel)
+			)
+
+			for _, eventLevel := range polyzero.Levels() {
+				funcSpy := testpolylog.FnMethodSpy{}
+				funcSpy.On("Fn", mock.AnythingOfType(polyzeroEventTypeName)).Return()
+
+				event := newTestEventWithLevel(t, logger, eventLevel)
+				expectedEventLevelEnabled := eventLevel.Int() >= loggerLevel.Int()
+
+				require.Equalf(t, expectedEventLevelEnabled, event.Enabled(), "expected event to be initially enabled")
+
+				// If the event level is GTE the logger level, then make additional
+				// assertions about #Func(), #Discard(), and #Enabled() behavior.
+				if expectedEventLevelEnabled {
+					// Assert that #Func() calls `funcSpy#Fn()` method 1 time with
+					// an event whose type name matches funcMethodEventTypeName.
+					event.Func(funcSpy.Fn)
+					funcSpy.AssertCalled(t, "Fn", mock.AnythingOfType(polyzeroEventTypeName))
```

Comment:
I understand your need for another mocking library now. I've seen this pattern in other languages.